### PR TITLE
updates to string formatting exercise

### PR DESCRIPTION
**Pull request by milesak60**
Updates made and built locally successfully. Not sure if this is now over the top, or needs more guidance but I tried to introduce some challenges.

Looks like from a quick scan through email the delivery method is changing, so I will take a look tomorrow, and contact Liz (et al).

### CODE REVIEW

**Comment by PythonCHB** [comment]
yeah, still confused about what's going to be published where / how, but this repo is the master copy for now.

**Comment by PythonCHB** [comment]
We're going to need to be careful about links back to Canvas, like:
 
 https://canvas.uw.edu/courses/1200526/assignments/3970003?module_item_id=7963708

I don't think those will stay static, and there will be multiple "instances" of the course.

And that link goes to the list exercise --not sure if that's intended.

File: source/exercises/string_formatting.rst [hunk 1]
```diff
@@ -130,15 +136,63 @@ Task Three
...
 
 
-Task four
+Task Five
 ---------
+f-strings are new to Python (version 3.6), but are very powerful and efficient. This means they are worth understanding and using. And this is made easier than it might be because they use the same, familiar formatting language that is conventionally used in Python (in .format()).
+
+So in this exercise we are going to specifically use f-strings.
+
+Here's the simplest example, just to show the basic syntax:
```
Comment (PythonCHB) [inline source/exercises/string_formatting.rst]: I'm not sure we should have an example if an f-string with no formatting -- kinda makes you wonder what the difference is -- and the answer is "nothing" :-)

So let's leave this out and go straight to the example with {name}

File: source/exercises/string_formatting.rst [hunk 2]
```diff
@@ -130,15 +136,63 @@ Task Three
...
+    Out[1]: 'a simple f-string'
+
+And here's an example that shows how you can use available variables in a f-string:
+
+.. code-block:: ipython
+
+    In [2]: name = 'Andy'
+    In [3]: f'Your name is {name}'
+    Out[3]: 'Your name is Andy'
+
```
Comment (PythonCHB) [inline source/exercises/string_formatting.rst]: Now let's add an example of evaluating an expression in an f-string:

In addition to referencing variables in the local scope, f-strings can evaluate simple expressions in line like so:

```
In [5]: f"Your name is {name.upper()}"
Out[5]: 'Your name is ANDY'

In [6]: name = "andy"

In [7]: f"Your name is {name.upper()}"
Out[7]: 'Your name is ANDY'
```

or 

```
In [8]: a = 5

In [9]: b = 10

In [10]: f"The sum is: {a+b}"
Out[10]: 'The sum is: 15'
```

File: source/exercises/string_formatting.rst [hunk 3]
```diff
@@ -130,15 +136,63 @@ Task Three
...
+* So here's a task for you. Given a four element list:
+
+    ``['oranges', 1.3, 'lemons', 1.1]``
+
+* Write an f-string that will display:
+
+    ``The weight of an orange is 1.3 and the weight of a lemon is 1.1``
+
+* Now see if you can change the f-string so that it displays the names of the fruit in upper case, and the weight 20% higher (that is 1.2 times higher).
```
Comment (PythonCHB) [inline source/exercises/string_formatting.rst]: nice!

File: source/exercises/string_formatting.rst [hunk 4]
```diff
@@ -130,15 +136,63 @@ Task Three
...
+One way to do that is:
+
+.. code-block:: ipython
+
+    '{:20}{:10}{:20}{:8}'.format('First', '$99.01', 'Second', '$88.09')
+
+In this simple example everything aligns nicely. But that will not be the case when the numbers to the left of the decimal place vary.
+Then you will need to use alignment specifiers. Do some research on this using the links below. Then:
+
+* Write some Python code to print a table of several rows, each with a name, an age and a cost. Make sure some of the costs are in the hunderds and thousands to test your alignment specifiers.
```
Comment (PythonCHB) [inline source/exercises/string_formatting.rst]: typo: hunderds => hundreds

**Comment by milesak60** [comment]
Great suggestions, thank you. Changes made and built successfully. Ready for another merge attempt!
________________________________

**Comment by PythonCHB** [comment]
Thanks!

**Pull request merged**